sky/data/mounting_utils.py

```diff
@@ -4,70 +4,104 @@
 
 
 def get_mounting_command(
+    mount_mode: storage_utils.StorageMode,
     mount_path: str,
-    install_cmd: str,
     mount_cmd: str,
+    install_cmd: Optional[str] = None,
     version_check_cmd: Optional[str] = None,
+    csync_log_path: Optional[str] = None,
 ) -> str:
     """Generates the mounting command for a given bucket.
 
-    Generated script first unmounts any existing mount at the mount path, checks
-    and installs the mounting utility if required, creates the mount path and
-    finally mounts the bucket.
+    There are two types of mounting supported in Skypilot, a traditional MOUNT
+    and CSYNC.
+
+    For traditional mounting, generated script first unmounts any
+    existing mount at the mount path, checks and installs the mounting utility
+    if required, creates the mount path and finally mounts the bucket.
+
+    For CSYNC, generated script first creates the CSYNC_PATH if it does not
+    exist, and finally runs CSYNC daemon on CSYNC_PATH to the bucket.
 
     Args:
+        mount_mode: Defines which mounting mode is used between traditional
+          MOUNT and CSYNC
         mount_path: Path to mount the bucket at.
-        install_cmd: Command to install the mounting utility. Should be
-          single line.
         mount_cmd: Command to mount the bucket. Should be single line.
+        install_cmd: Command to install the mounting utility for MOUNT mode.
+          Should be single line.
 
     Returns:
         str: Mounting command with the mounting script as a heredoc.
     """
     mount_binary = mount_cmd.split()[0]
     installed_check = f'[ -x "$(command -v {mount_binary})" ]'
-    if version_check_cmd is not None:
-        installed_check += f' && {version_check_cmd}'
+    if mount_mode == storage_utils.StorageMode.MOUNT:
+        script_path = f'~/.sky/mount_{random.randint(0, 1000000)}.sh'
+        assert csync_log_path is None, ('CSYNC log path should '
+                                        'not be defined for MOUNT mode.')
+        if version_check_cmd is not None:
+            installed_check += f' && {version_check_cmd}'
+    else:  # script path for CSYNC mode
+        script_path = f'~/.sky/csync_{random.randint(0, 1000000)}.sh'
+        assert install_cmd is None, ('Installing commands should '
+                                     'not be defined for CSYNC mode.')
+
     script = textwrap.dedent(f"""
         #!/usr/bin/env bash
         set -e
 
-        MOUNT_PATH={mount_path}
-        MOUNT_BINARY={mount_binary}
+        MOUNT_MODE={mount_mode.value}
+        MOUNT_PATH='{mount_path}'
+        echo "MOUNT_MODE is: $MOUNT_MODE"
 
-        # Check if path is already mounted
-        if grep -q $MOUNT_PATH /proc/mounts ; then
-            echo "Path already mounted - unmounting..."
-            fusermount -uz "$MOUNT_PATH"
-            echo "Successfully unmounted $MOUNT_PATH."
-        fi
+        if [ "$MOUNT_MODE" = "MOUNT" ]; then
+            MOUNT_BINARY={mount_binary}
```

Comment:
Can we move this to the top of the file? Reason: The scope of this variable is a little bit confusing.